src/components/utils/DueDateLayout.tsx

```diff
@@ -1,21 +1,28 @@
 import { Typography } from '@mui/material'
-import { format, parseISO, isToday, isTomorrow } from 'date-fns'
+import dayjs from 'dayjs'
+import isToday from 'dayjs/plugin/isToday'
+import isTomorrow from 'dayjs/plugin/isTomorrow'
+
+dayjs.extend(isToday)
+dayjs.extend(isTomorrow)
 
 interface DueDateLayoutProp {
   date: string | Date
 }
 
 export const DueDateLayout = ({ date }: DueDateLayoutProp) => {
-  const now = new Date()
+  const now = dayjs()
   let formattedDueDate
   let isPast: boolean
-  if (isToday(date)) {
+  const parsedDate = dayjs(date)
+
+  if (parsedDate.isToday()) {
     formattedDueDate = `Today`
-  } else if (isTomorrow(date)) {
+  } else if (parsedDate.isTomorrow()) {
     formattedDueDate = `Tomorrow`
   } else {
-    isPast = now > new Date(date)
-    formattedDueDate = format(date, 'dd, MMM, yyyy')
+    isPast = now.isAfter(parsedDate)
+    formattedDueDate = parsedDate.format('MMMM D, YYYY')
   }
 
```

Comment:
Let's not create a `utils` inside components. IMO, there shouldn't exist utility components. Components that performs logical operations before rendering should be higher order components and should be placed inside `hoc` folder. The date related code that you've written here should be inside an `hoc` which should pass the computed value to their child.

Reply:
I think it is not feasible to create a higher order component, however higher order functions works good. But we cannot use hooks for computations in higher order functions. What are your thoughts on this?